data/data/azure/dns/dns.tf

```diff
@@ -32,6 +32,8 @@ resource "azureprivatedns_a_record" "api_internal" {
 }
 
 resource "azurerm_dns_cname_record" "api_external" {
+  count = var.preexisting_network ? 0 : 1
```

Comment:
Why are we not creating the API record when there is preexisting network?

Reply:
Fixed.